var/spack/repos/builtin/packages/py-tensorflow/package.py

```diff
@@ -138,38 +144,51 @@ class PyTensorflow(Package, CudaPackage):
     depends_on('py-future', type='build', when='^python@:2')
 
     # Listed under REQUIRED_PACKAGES in tensorflow/tools/pip_package/setup.py
-    depends_on('py-absl-py@0.10:0', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-absl-py@0.4:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-absl-py@0.10:0', type=('build', 'run'), when='@2.4.0:2.6')
     depends_on('py-absl-py@0.7.0:', type=('build', 'run'), when='@1.12.1,1.14:2.3')
     depends_on('py-absl-py@0.1.6:', type=('build', 'run'), when='@1.5:1.11')
-    depends_on('py-astunparse@1.6.3:1.6', type=('build', 'run'), when='@2.4.0:')
+    depends_on('py-astunparse@1.6.0:', type=('build', 'run'), when='@2.7:')
+    depends_on('py-astunparse@1.6.3:1.6', type=('build', 'run'), when='@2.4.0:2.6')
     depends_on('py-astunparse@1.6.3', type=('build', 'run'), when='@2.2:2.3')
     depends_on('py-astor@0.6.0:', type=('build', 'run'), when='@1.6:2.1')
     depends_on('py-backports-weakref@1.0:', type=('build', 'run'), when='@1.3: ^python@:3.3')
     depends_on('py-backports-weakref@1.0rc1', type=('build', 'run'), when='@1.2.0:1.2.1')
+    depends_on('py-clang@9.0.1:', type=('build', 'run'), when='@2.7:')
```

Comment:
Well, this is interesting. For tensorflow-2.6:
```
'clang ~= 5.0'
```
For tensorflow-2.7:
```
'libclang >= 9.0.1'
```
They are two different projects on PyPI, although I think they are compatible for common versions. Since I started this with tensorflow-2.6, I used the clang project. It runs from llvm-5 to llvm-11. The libclang project runs from llvm-9 to llvm-12. 

I could create a new py-libclang package, but that means that py-clang would only exist as a dependency for py-tensorflow-2.6. I do not see another way as libclang would not meet the requirements for tensorflow-2.6. I am worried about confusion between py-clang and py-libclang though. 

Perhaps the best option is to limit the versions of py-clang from 5 to 8, and have py-libclang from 9 and up, eliminating the crossover between versions 9 to 11. Perhaps there could be a virtual py-clang package.